src/cloudsc_fortran/dwarf_cloudsc.F90

```diff
@@ -68,8 +81,11 @@ PROGRAM DWARF_CLOUDSC
 ! TODO: Create a global global memory state from serialized input data
 CALL GLOBAL_STATE%LOAD(NPROMA, NGPTOT, NGPTOTG)
 
+CALL INITIALIZE_PARAMETER_DATATYPES(YDOMCST, YDOETHF, YDECLDP, YDEPHLI )
+
 ! Call the driver to perform the parallel loop over our kernel
-CALL CLOUDSC_DRIVER(NUMOMP, NPROMA, GLOBAL_STATE%KLEV, NGPTOT, NGPTOTG, &
+CALL CLOUDSC_DRIVER( NUMOMP, NPROMA, GLOBAL_STATE%KLEV, NGPTOT, NGPTOTG, &
+     & NCLDQV, NCLDQL, NCLDQR, NCLDQI, NCLDQS, NCLV,  &
```

Comment:
The global cloud vector indexes/sizes can probably also be wrapped in a derived type to avoid passing all of them down.

Reply:
Is derived type necessary or it could be just a vector ? Perhaps it could be hardcoded (e.g. called integer :: TNCLV(NCLV)) in MODULE YOECLDP ?

Reply:
These are named integer indexes. Why would we want to create yet another pattern, instead of using the same as for the other module-level parameters? Passing them as a derived type object we can use associates in the subroutine body, following the existing convection, no?

Reply:
@mlange I understand that explicit declaration of NCLDQV, NCLDQL, NCLDQR, NCLDQI, NCLDQS, NCLV must stay as they are in the yoecldp, without putting them in the derived type envelope. So the information needs to  be duplicated somewhere, what about putting them as members of TECLDP, so no extra type definition is need ?

Reply:
Moreover, in the current form they can be resolved at compile time (since these are parameters). If I include them as variables in the derived type, then it won't be possible anymore. So repeating them in the derived type as parameters seems to be desirable option.

Reply:
Yes, adding them to TECLDP and setting them in the initialisation routine should work nicely I think.